### PR TITLE
support incremental for fragments

### PR DESCRIPTION
**Pull request by superyyrrzz**

### CODE REVIEW

File: test/Microsoft.DocAsCode.Build.SchemaDriven.Tests/MergeMarkdownFragmentsTest.cs [hunk 1]
```diff
@@ -267,6 +267,42 @@ markdown content
             Assert.Null(rawModel["summary"]);
         }
 
+        [Fact]
+        public void TestFragmentsWithIncremental()
+        {
+            // first build
```
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.SchemaDriven.Tests/MergeMarkdownFragmentsTest.cs]: use test log listener to validate logs when incremental build

File: src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaDrivenDocumentProcessor.cs [hunk 2]
```diff
@@ -141,28 +141,25 @@ public override FileModel Load(FileAndType file, ImmutableDictionary<string, obj
                         {
                             LocalPathFromRoot = localPathFromRoot,
                         };
+                        fm.MarkdownFragmentsModel = new FileModel(
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaDrivenDocumentProcessor.cs]: why we still need create FragmentsModel when the content is null?

Comment (superyyrrzz) [reply src/Microsoft.DocAsCode.Build.SchemaDriven/SchemaDrivenDocumentProcessor.cs]: This model is used to report a dependency. Even md doesn't exist, DocFx should think yaml has a dependency on it.

File: test/Microsoft.DocAsCode.Build.SchemaDriven.Tests/MergeMarkdownFragmentsTest.cs [hunk 3]
```diff
@@ -267,6 +267,78 @@ markdown content
...
+                {
+                    BuildDocument(_files);
+                }
+                Assert.True(File.Exists(_rawModelFilePath));
+                rawModel = JsonUtility.Deserialize<JObject>(_rawModelFilePath);
+                Assert.NotNull(rawModel["summary"]);
+                Assert.Contains("I update a summary", rawModel["summary"].ToString());
+                Assert.NotNull(listener.Items.FirstOrDefault(s => s.Message.StartsWith("There is no template processing document type(s): RESTMixedTest")));
+                Assert.NotNull(listener.Items.FirstOrDefault(s => s.Message.StartsWith("Can't find") && s.Message.EndsWith("/invalid.md.")));
+                listener.Items.Clear();
```
Comment (vicancy) [inline test/Microsoft.DocAsCode.Build.SchemaDriven.Tests/MergeMarkdownFragmentsTest.cs]: Add a check to compare logs from each round. They should be the same.

**Pull request merged**